source/unified-test-format/unified-test-format.rst

```diff
@@ -2493,7 +2559,229 @@ An example of this operation follows::
         client: *client0
         connections: 1
 
+runOnThread
+~~~~~~~~~~~
+
+The "runOnThread" operation instructs the test runner to schedule an operation
+to be run on a given thread. runOnThread MUST NOT wait for the scheduled
+operation to complete. If any of the operation's test assertions fail, the
+entire test case MUST fail as well.
+
+The following arguments are supported:
+
+- ``thread``: Required string. Thread entity on which this operation should be
+  scheduled.
+
+- ``operation``: Required `operation`_ object. The operation to schedule on the
+  thread. This object must be a valid operation as described in `Entity Test
+  Operations`_.
+
+An example of this operation follows::
+
+     - name: runOnThread
+       object: testRunner
+       arguments:
+         thread: *thread0
+         operation:
+           name: insertOne
+           object: *collection0
+           arguments:
+             document: { _id: 2 }
+           expectResult:
+             $$unsetOrMatches:
+               insertedId: { $$unsetOrMatches: 2 }
+
+
+waitForThread
+~~~~~~~~~~~~~
+
+The "waitForThread" operation instructs the test runner to notify the given
+thread that no more operations are forthcoming, wait for it to complete its last
+operation, and assert that it exited without any errors.
```

Comment:
Responded to this above, but yes I think we probably should. Will update the text to reflect this when I get back from PTO.

Reply:
More precisely, we can note under `runOnThread` that those operations SHOULD be followed by a `waitForThread` operation later in the `test.operations` list. Using "SHOULD" instead of "MUST" since this is guidance for writing test files and not something we can (yet) enforce in the schema (see: [Design Rationale](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#design-rationale)).

Reply:
Added a line about this. I ended up putting it with `waitForThread`, since it seemed more relevant here and also because it's technically possible we add other thread operations that may be similarly checked for errors via waitForThread.